samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -323,10 +441,102 @@ def _enrich_mapped_resources(
+        cfn_resource["Metadata"]["WorkingDirectory"] = terraform_application_dir
+        # currently we set the terraform project root directory that contains all the terraform artifacts as the project
+        # directory till we work on the custom hook properties, and add a property for this value.
+        cfn_resource["Metadata"]["ProjectRootDirectory"] = terraform_application_dir
 
     def _enrich_image_lambda_function(sam_metadata_resource: Dict, cfn_resource: Dict, cfn_resource_logical_id: str):
-        pass
+        sam_metadata_attributes = sam_metadata_resource.get("values", {}).get("triggers", {})
+        sam_metadata_resource_address = sam_metadata_resource.get("address")
+        if not sam_metadata_resource_address:
+            raise PrepareHookException(
+                "Invalid Terraform plan output. The address property should not be null to any terraform resource."
+            )
+        cfn_resource_properties = cfn_resource.get("Properties", {})
+
+        LOG.info(
+            "Enrich the IMAGE lambda function %s using the metadata properties defined in resource %s",
+            cfn_resource_logical_id,
+            sam_metadata_resource_address,
+        )
+
+        _validate_referenced_resource_matches_sam_metadata_type(
+            cfn_resource, sam_metadata_attributes, sam_metadata_resource_address, IMAGE
+        )
+
+        cfn_docker_context_path = _get_lambda_function_source_code_path(
+            sam_metadata_attributes,
+            sam_metadata_resource_address,
+            terraform_application_dir,
+            "docker_context",
+            "docker_context_property_path",
+            "docker context",
+        )
+        cfn_docker_file = sam_metadata_attributes.get("docker_file")
+        cfn_docker_build_args_string = sam_metadata_attributes.get("docker_build_args")
+        if cfn_docker_build_args_string:
+            try:
+                LOG.debug("Parse the docker build args %s", cfn_docker_build_args_string)
+                cfn_docker_build_args = json.loads(cfn_docker_build_args_string)
+                if not isinstance(cfn_docker_build_args, dict):
+                    raise InvalidSamMetadataPropertiesException(
+                        f"The sam metadata resource {sam_metadata_resource_address} should contain a valid json "
+                        f"encoded string for the lambda function docker build arguments."
+                    )
+            except Exception as exc:
+                raise InvalidSamMetadataPropertiesException(
+                    f"The sam metadata resource {sam_metadata_resource_address} should contain a valid json encoded "
+                    f"string for the lambda function docker build arguments."
+                ) from exc
+
+        cfn_docker_tag = sam_metadata_attributes.get("docker_tag")
+
+        if cfn_resource_properties.get("Code"):
+            cfn_resource_properties.pop("Code")
```

Comment:
why do we do this?

Reply:
We do not need the Code attribute where we already mapped the image uri. This will be added during the build step.

Reply:
hmmm but do we need to remove it or are you just removing it because it won't be used? Could we add a comment in the code with an explanation maybe, in case somebody else wonders this?

Reply:
We need to remove the `Code.IamgeUri` property for the Lambda function as SAM CLI will use theImage Build metadata properties to build a local image uri for this function to be used in local testing. I do not want SAM CLI to use the provided image uri, and so the customer will not be able to change its code, and then test it locally.

Reply:
let me add some comment to explain that in the code